src/Symfony/Component/Translation/Translator.php

```diff
@@ -171,7 +171,11 @@ public function setLocale($locale)
      */
     public function getLocale()
     {
-        return $this->locale;
+        if (null === $this->locale) {
+            @trigger_error(sprintf('Using a "null" $locale is deprecated since Symfony 4.4. You should set it via %s::setLocale($locale) explicitly.', static::class, \E_USER_DEPRECATED));
+        }
+
+        return $this->locale ?? 'en';
```

Comment:
btw TranslatorTrait does `return $this->locale ?: \Locale::getDefault();`, seems reasonable to keep it consistent :/

Reply:
But this means an empty locale (`$locale = ''`) is valid as well. Not sure this is the case? Tests are failing if I use `return $this->locale ?: \Locale::getDefault();`.

Reply:
binding resources to an empty locale seems valid:

https://github.com/symfony/symfony/blob/516fa40c34fcbc38faea1b9993e5e0981eb5c991/src/Symfony/Component/Translation/Tests/TranslatorTest.php#L622-L625

so `??` is the right approach i figure, which raises questions why the trait uses `?:`, but that's out of scope here i guess :sweat_smile: